src/js/switcher_tabs.js

```diff
@@ -0,0 +1,72 @@
+import { Dimensions } from "react-native"
+import { interpolate, useAnimatedGestureHandler, useAnimatedScrollHandler, useAnimatedStyle, withSpring  } from "react-native-reanimated"
+
+export const SCREEN_WIDTH = Dimensions.get('window').width
+
+export function useSwitcherTabsGestureHandler(translateX) {
+  return useAnimatedGestureHandler({
+    onStart: (_event, context) => {
+      context.startX = translateX.value
+    },
+    onActive: (gestureEvent, context) => {
+      translateX.value = context.startX + gestureEvent.translationX
+    },
+    onEnd: () => {
+      translateX.value = withSpring(0)
+    }
+  })
+}
+
+export function useSwitcherTabsStyle(translateX) {
+  return useAnimatedStyle(() => {
+    'worklet'
+
+    return {
+      transform: [
+        {
+          translateX: translateX.value
+        }
+      ]
+
+    }
+  })
+}
```

Comment:
[apply-animations-to-style](https://github.com/status-im/status-mobile/blob/a94797710a98408144a86764fd78013b1d72e1c2/src/quo2/reanimated.cljs#L56) function can be used to create animated style worklet in cljs instead of `useSwitcherTabsStyle`

Reply:
How would i use `apply-animations-to-style`, if i do arithmetic operations with the shared value first?

Like here [here](https://github.com/chrispader/status-mobile/blob/b0f575374a7093990590948165b5e20a247a09c1/src/js/switcher_tabs.js#L68) or [here](https://github.com/chrispader/status-mobile/blob/b0f575374a7093990590948165b5e20a247a09c1/src/js/switcher_tabs.js#L50)

Reply:
Something like this

```cljs
 (let [{:keys [width]} (constants/dimensions)
           header-text-style {:color "#FFFFFF"}
           animated-style       (reanimated/apply-animations-to-style
                                 {:opacity (reanimated/interpolate (/ translate-x width) [(- tab-index 1) tab-index (+ tab-index 1)] [0.5 1 0.5] (clj->js {:extrapolate-left "clamp" :extrapolate-right "clamp"}))}
                                 header-text-style)]
```

is not working for me, because `.value` is accessed inside the worklet factory, so i would have to provide a shared value

Reply:
> How would i use apply-animations-to-style, if i do arithmetic operations with the shared value first?

You can apply arithmetic options to shared values and derive new shared values using the [useDerivedValue](https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useDerivedValue) hook
Ex:
[Derivation](https://github.com/status-im/status-mobile/blob/b565cbf85e496dd08c73230d45fe24d90b989c77/src/js/worklet_factory.js#L44), [Import](https://github.com/status-im/status-mobile/blob/b565cbf85e496dd08c73230d45fe24d90b989c77/src/status_im/switcher/animation.cljs#L36), [Use](https://github.com/status-im/status-mobile/blob/b565cbf85e496dd08c73230d45fe24d90b989c77/src/status_im/switcher/switcher.cljs#L64)

Reply:
Ok, i understand, so this will still need to be in plain JS. I was searching for a way to do this in CLJS, but then extracted all of the REA parts to JS